src/BrokerToken.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import '@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol';
+
+/**
+ * @notice Broker and Canary utility token is a simple ERC20 token with permit functionality.
+ * All the supply is minted to the deployer.
+ */
+contract BrokerToken is ERC20Permit {
+	/**
+	 * @dev Simple constructor, passing arguments to ERC20Permit and ERC20 constructors.
+	 * Mints the supply to the deployer.
+	 * @param name Name of the Token.
+	 * @param symbol Symbol of the Token.
+	 * @param supply Maximum supply of the Token.
+	 */
+	constructor(
+		string memory name,
+		string memory symbol,
+		uint256 supply
+	) ERC20Permit(name) ERC20(name, symbol) {
+		_mint(msg.sender, supply);
```

Comment:
Maybe we need to do supply x 10^decimals ?
or mint does it ?

Reply:
No, `_mint` does not do this for us. Usually the user does the math himself